### PR TITLE
Add an ignition plugin for latching simulation

### PR DESCRIPTION
**Pull request by ibrahiminfinite**
This PR adds an ignition gazebo latching plugin to simulate the latching mechanism for robot to latch the carts.

The plugin provides an ignition topic interface to activate/deactivate the latching mechanism, which is then bridged to ROS2

To activate the latch: `ros2 topic pub fake_latch/latch std_msgs/msg/String 'data: "<cart_name>"'`
To deactivate the latch: `ros2 topic pub fake_latch/unlatch std_msgs/msg/String 'data: "<cart_name>"'`

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,32 @@ find_package(ament_cmake_python REQUIRED)
...
+find_package(ignition-msgs8 REQUIRED)
+find_package(ignition-plugin1 REQUIRED COMPONENTS loader register)
+set(IGN_PLUGIN_VER ${ignition-plugin1_VERSION_MAJOR})
+
+
+# Add fake latch plugin lib
+include_directories(
+  include
+)
+add_library(FakeLatchPlugin SHARED src/fake_latch_plugin.cpp)
```
Comment (AndyZe) [inline CMakeLists.txt]:
```suggestion
include_directories(
  include
)

# Add fake latch plugin lib
add_library(FakeLatchPlugin SHARED src/fake_latch_plugin.cpp)
```

File: description/rover_2wd.urdf [hunk 2]
```diff
@@ -376,6 +376,14 @@
     </plugin>
   </gazebo>
 
+
+  <gazebo>
```
Comment (AndyZe) [inline description/rover_2wd.urdf:381]:
```suggestion
  <!-- Cart latching -->
  <gazebo>
```

File: include/nav2_gazebo/fake_latch_plugin.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#pragma once
```
Comment (AndyZe) [inline include/nav2_gazebo/fake_latch_plugin.hpp:1]: Indentations are 4 spaces here. Usually we use 2 for C++

File: include/nav2_gazebo/fake_latch_plugin.hpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            void PreUpdate([[maybe_unused]] const ignition::gazebo::UpdateInfo &_nfo,
+                ignition::gazebo::EntityComponentManager &ecm) override;
+
+            void LatchMsgCallback(const ignition::msgs::StringMsg &latch_msg);
+
+            void UnlatchMsgCallback(const ignition::msgs::StringMsg &unlatch_msg);
+
+        private:
+
+
```
Comment (AndyZe) [inline include/nav2_gazebo/fake_latch_plugin.hpp:42]:
```suggestion
```

File: src/fake_latch_plugin.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
+#include <nav2_gazebo/fake_latch_plugin.hpp>
+#include <ignition/gazebo/Model.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/gazebo/components.hh>
+#include <ignition/gazebo/components/DetachableJoint.hh>
+
+namespace nav2_gazebo
+{
+
+    FakeLatchPlugin::FakeLatchPlugin()
```
Comment (AndyZe) [inline src/fake_latch_plugin.cpp:10]: Use 2-space indentation here, too

File: src/fake_latch_plugin.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            robot_name_ = ecm.Component<ignition::gazebo::components::Name>(entity)->Data();
+            ignwarn << "FakeLatchPlugin has been configured for robot: " << robot_name_ << std::endl;
+        }
+
+    }
+
+    void FakeLatchPlugin::PreUpdate([[maybe_unused]] const ignition::gazebo::UpdateInfo &info,
+        ignition::gazebo::EntityComponentManager &ecm)    
+    {
+
```
Comment (AndyZe) [inline src/fake_latch_plugin.cpp:54]:
```suggestion
```

File: src/fake_latch_plugin.cpp [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    FakeLatchPlugin::FakeLatchPlugin()
+    : latched_{false}
+    , unlatch_requested_{false}
+    , robot_name_{""}
+    , cart_name_{""}
+    , robot_{ignition::gazebo::kNullEntity}
+    , cart_{ignition::gazebo::kNullEntity}
+    , latch_joint_{ignition::gazebo::kNullEntity}
+    {
+        ignwarn << "Initializing FakeLatch plugin" << std::endl;
```
Comment (AndyZe) [inline src/fake_latch_plugin.cpp]: Get rid of this? Might be more annoying than helpful.

**Review by AndyZe — COMMENTED** [review]
Just some formatting stuff.

We prob should get a clang-format job running on this repo. I'll put it on the TODO list.

File: src/fake_latch_plugin.cpp (hunk 7 above)
Comment (ibrahiminfinite) [reply src/fake_latch_plugin.cpp]: Yeah this can be removed

**Review by AndyZe — APPROVED** [review]

**Pull request merged**